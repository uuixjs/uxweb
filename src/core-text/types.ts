/**
 * Many string values in this file must match the CSS values
 * for the corresponding property because they get used directly!
 */

export type Lines = 1 | 2;

export enum LineHeight {
  Body = "body",
  Heading = "heading",
}

export type LineHeightValue = "body" | "heading"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum TextDecoration {
  Underline = "underline",
  Strikethrough = "line-through",
}

export type TextDecorationValue = "underline" | "line-through"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum TextTransform {
  Capitalize = "capitalize",
  Uppercase = "uppercase",
}

export type TextTransformValue = "capitalize" | "uppercase"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum TextType {
  P = "p",
  Span = "span",
  H1 = "h1",
  H2 = "h2",
  H3 = "h3",
  H4 = "h4",
  H5 = "h5",
  H6 = "h6",
  Strong = "strong",
  Em = "em",
}

export type TextTypeValue =
  | "p"
  | "span"
  | "h1"
  | "h2"
  | "h3"
  | "h4"
  | "h5"
  | "h6"
  | "strong"
  | "em"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum WordBreak {
  Normal = "normal",
  BreakAll = "break-all",
  KeepAll = "keep-all",
  BreakWord = "break-word",
}

export type WordBreakValue = "normal" | "break-all" | "keep-all" | "break-word"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum WhiteSpace {
  NoWrap = "nowrap",
  PreWrap = "pre-wrap",
}

export type WhiteSpaceValue = "nowrap" | "pre-wrap"; // TODO: https://jira.xarth.tv/browse/COREUI-3554
