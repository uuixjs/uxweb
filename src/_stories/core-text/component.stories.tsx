import * as React from "react";

import {
  Background,
  Color,
  Column,
  CoreText,
  FontSize,
  FontWeight,
  Grid,
  InjectLayout,
  Layout,
  LineHeight,
  Overflow,
  TextDecoration,
  TextTransform,
  TextType,
  VerticalAlign,
  WhiteSpace,
  WordBreak,
} from "v2";

import { Enum } from "lib";

export default { title: "CoreText" };

const loremIpsum =
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.";

export const ellipsis = () => (
  <Grid>
    <Column cols={6}>
      <Layout padding={1} fullHeight border>
        <CoreText bold>One Line</CoreText>
        <CoreText ellipsis>{loremIpsum}</CoreText>
      </Layout>
    </Column>
    <Column cols={6}>
      <Layout padding={1} fullHeight border>
        <CoreText bold>Two Lines</CoreText>
        <CoreText ellipsis lines={2}>
          {loremIpsum}
        </CoreText>
      </Layout>
    </Column>
  </Grid>
);

export const textWrap = () => (
  <Grid>
    <Column cols={6}>
      <Layout padding={1} fullHeight border>
        <CoreText bold>Pre-Wrap</CoreText>
        <CoreText whiteSpace={WhiteSpace.PreWrap}>
          These line breaks should be preserved:
          {`
Line One
Line Two

Line Four
Line Five

Line Seven

`}
          {loremIpsum}
        </CoreText>
      </Layout>
    </Column>
    <Column cols={6}>
      <Layout padding={1} overflow={Overflow.Hidden} border>
        <CoreText bold>No Wrap</CoreText>
        <CoreText whiteSpace={WhiteSpace.NoWrap}>{loremIpsum}</CoreText>
      </Layout>
    </Column>
  </Grid>
);

export const tabularNums = () => (
  <Layout padding={1} fullHeight border>
    <table>
      <tr>
        <td>Without tabular nums: &nbsp;</td>
        <td>
          <CoreText>1234</CoreText>
        </td>
      </tr>
      <tr>
        <td>With tabular nums: &nbsp;</td>
        <td>
          <CoreText tabularNums>1234</CoreText>
        </td>
      </tr>
    </table>
  </Layout>
);

export const textStyling = () => (
  <>
    <Layout padding={{ y: 1, x: 2 }} margin={{ bottom: 2 }} border>
      {Enum.keys(TextType).map((key) => (
        <Layout key={key} margin={{ y: 0.5 }}>
          <CoreText type={TextType[key]}>TextType.{key}</CoreText>
        </Layout>
      ))}
    </Layout>

    <Layout padding={{ y: 1, x: 2 }} margin={{ bottom: 2 }} border>
      {Enum.keys(FontSize).map((key) => (
        <Layout key={key} margin={{ y: 0.5 }}>
          <CoreText fontSize={FontSize[key]}>FontSize.{key}</CoreText>
        </Layout>
      ))}
    </Layout>

    <Layout padding={{ y: 1, x: 2 }} margin={{ bottom: 2 }} border>
      {Enum.keys(Color).map((key) => (
        <Layout key={key} margin={{ y: 0.5 }}>
          <CoreText color={Color[key]}>Color.{key}</CoreText>
        </Layout>
      ))}
    </Layout>

    <Layout padding={{ y: 1, x: 2 }} margin={{ bottom: 2 }} border>
      {Enum.keys(FontWeight).map((key) => (
        <Layout key={key} margin={{ y: 0.5 }}>
          <CoreText fontWeight={FontWeight[key]}>FontWeight.{key}</CoreText>
        </Layout>
      ))}
    </Layout>

    <Layout padding={{ y: 1, x: 2 }} margin={{ bottom: 2 }} border>
      {Enum.keys(TextDecoration).map((key) => (
        <Layout key={key} margin={{ y: 0.5 }}>
          <CoreText decoration={TextDecoration[key]}>
            TextDecoration.{key}
          </CoreText>
        </Layout>
      ))}
    </Layout>

    <Layout padding={{ y: 1, x: 2 }} margin={{ bottom: 2 }} border>
      <Layout margin={{ y: 0.5 }}>
        <CoreText italic>italic</CoreText>
      </Layout>
      <Layout margin={{ y: 0.5 }}>
        <CoreText bold>bold</CoreText>
      </Layout>
    </Layout>
  </>
);

export const textTransformations = () => (
  <>
    <Layout padding={{ y: 1, x: 2 }} margin={{ bottom: 2 }} border>
      {Enum.keys(TextTransform).map((key) => (
        <Layout key={key} margin={{ y: 0.5 }}>
          <CoreText bold>TextTransform.{key}:</CoreText>
          <CoreText transform={TextTransform[key]}>
            The quick brown fox jumps over the lazy dog
          </CoreText>
        </Layout>
      ))}
    </Layout>

    <Layout padding={{ y: 1, x: 2 }} margin={{ bottom: 2 }} border>
      {Enum.keys(LineHeight).map((key) => (
        <Layout key={key} margin={{ y: 0.5 }}>
          <CoreText bold>LineHeight.{key}:</CoreText>
          <CoreText lineHeight={LineHeight[key]}>{loremIpsum}</CoreText>
        </Layout>
      ))}
    </Layout>

    <Layout padding={{ y: 1, x: 2 }} margin={{ bottom: 2 }} border>
      {Enum.keys(WordBreak).map((key) => (
        <Layout key={key} margin={{ top: 0.5, bottom: 1 }}>
          <CoreText bold>WordBreak.{key}:</CoreText>
          <InjectLayout border>
            <div style={{ width: "20rem" }}>
              <CoreText wordBreak={WordBreak[key]}>
                Honorificabilitudinitatibus califragilisticexpialidocious
                Taumatawhakatangihangakoauauotamateaturipukakapikimaungahoronukupokaiwhenuakitanatahu
                次の単語グレートブリテンおよび北アイルランド連合王国で本当に大きな言葉
              </CoreText>
            </div>
          </InjectLayout>
        </Layout>
      ))}
    </Layout>

    <Layout padding={{ y: 1, x: 2 }} margin={{ bottom: 2 }} border>
      {Enum.keys(VerticalAlign).map((key) => (
        <Layout key={key} margin={{ top: 0.5, bottom: 1 }}>
          <Layout borderTop borderBottom>
            <InjectLayout
              verticalAlign={VerticalAlign[key]}
              background={Background.Alt}
              margin={{ right: 1 }}
            >
              <img src="" width="50" height="50" />
            </InjectLayout>
            <CoreText
              align={VerticalAlign[key]}
              type={TextType.Span}
              fontSize={FontSize.Size6}
            >
              VerticalAlign.{key}
            </CoreText>
          </Layout>
        </Layout>
      ))}
    </Layout>
  </>
);
