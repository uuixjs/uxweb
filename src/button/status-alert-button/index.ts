export { StatusAlertButton, StatusAlertButtonProps } from "./component";
