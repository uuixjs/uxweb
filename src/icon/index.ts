export { Icon, IconProps } from "./component";
