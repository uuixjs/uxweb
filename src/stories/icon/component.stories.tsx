import * as React from "react";

import {
  Background,
  Color,
  Column,
  CoreText,
  Grid,
  Icon,
  Layout,
  SVG,
  SVGAsset,
  TextType,
} from "v2";

import { Component } from "react";

export default { title: Icon.displayName };

export const examples = () => <IconPage />;

class IconPage extends Component {
  public render() {
    const icons = (
      <>
        <Icon asset={SVGAsset.Heart} />
        <Icon asset={SVGAsset.GlyphLive} />
        <Icon asset={SVGAsset.NavGames} />
        <Grid>
          <Column cols={6}>
            <CoreText type={TextType.H5}>Icon using Aspect Fill</CoreText>
            <Layout background={Background.Alt2}>
              <Icon asset={SVGAsset.Unheart} fill />
            </Layout>
          </Column>
          <Column cols={6}>
            <CoreText type={TextType.H5}>
              SVG using SVG width/height=100%
            </CoreText>
            <Layout background={Background.Alt2}>
              <SVG asset={SVGAsset.Heart} fill />
            </Layout>
          </Column>
        </Grid>
      </>
    );

    return (
      <Layout margin={{ bottom: 2 }}>
        <Layout
          padding={3}
          margin={{ y: 2 }}
          background={Background.Alt}
          border
        >
          <CoreText>With no color set:</CoreText>
          {icons}
        </Layout>

        <Layout
          padding={3}
          margin={{ y: 2 }}
          background={Background.Alt}
          color={Color.Link}
          border
        >
          <CoreText>With link color set:</CoreText>
          {icons}
        </Layout>
      </Layout>
    );
  }
}
