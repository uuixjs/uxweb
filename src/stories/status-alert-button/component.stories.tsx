import * as React from "react";

import {
  Button,
  CoreButtonSize,
  Layout,
  SVGAsset,
  StatusAlertButton,
} from "v2";
import {
  CombinationGenerator,
  CombinationMode,
} from "../../components/combination-generator";

export default { title: "StatusAlertButton" };

export const combinations = () => (
  <>
    <CombinationGenerator
      mode={CombinationMode.Simple}
      fields={[
        {
          propKey: "fullWidth",
          propValues: [false, true],
        },
        {
          propKey: "disabled",
          propValues: [false, true],
        },
      ]}
    >
      <Button>Star</Button>
      <Button icon={SVGAsset.Star} statusAlertIcon={SVGAsset.StarHollow}>
        Star
      </Button>
      <Button
        icon={SVGAsset.Star}
        statusAlertIcon={SVGAsset.StarHollow}
        statusAlertText="Unstar"
      >
        Star
      </Button>
      <Button statusAlertText="Unstar">Star</Button>
    </CombinationGenerator>
    <CombinationGenerator
      mode={CombinationMode.Simple}
      fields={[
        {
          propKey: "fullWidth",
          propValues: [false, true],
        },
        {
          propKey: "disabled",
          propValues: [false, true],
        },
        {
          propKey: "size",
          propEnum: CoreButtonSize,
          propEnumName: "CoreButtonSize",
        },
      ]}
    >
      <StatusAlertButton>Star</StatusAlertButton>
      <StatusAlertButton
        icon={SVGAsset.Star}
        statusAlertIcon={SVGAsset.StarHollow}
      />
      <StatusAlertButton
        icon={SVGAsset.Star}
        statusAlertIcon={SVGAsset.StarHollow}
      >
        Star
      </StatusAlertButton>
      <StatusAlertButton
        icon={SVGAsset.Star}
        statusAlertIcon={SVGAsset.StarHollow}
        statusAlertText="Unstar"
      >
        Star
      </StatusAlertButton>
      <StatusAlertButton statusAlertText="Unstar">Star</StatusAlertButton>
    </CombinationGenerator>
    <CombinationGenerator
      mode={CombinationMode.Simple}
      fields={[
        {
          propKey: "fullWidth",
          propValues: [false, true],
        },
        {
          propKey: "disabled",
          propValues: [false, true],
        },
        {
          propKey: "size",
          propValues: [
            CoreButtonSize.Small,
            CoreButtonSize.Default,
            CoreButtonSize.Large,
          ],
        },
      ]}
    >
      <StatusAlertButton showAlert>Star</StatusAlertButton>
      <StatusAlertButton
        showAlert
        icon={SVGAsset.Star}
        statusAlertIcon={SVGAsset.StarHollow}
      >
        Star
      </StatusAlertButton>
      <StatusAlertButton
        showAlert
        icon={SVGAsset.Star}
        statusAlertIcon={SVGAsset.StarHollow}
        statusAlertText="Unstar"
      >
        Star
      </StatusAlertButton>
      <StatusAlertButton showAlert statusAlertText="Unstar">
        Star
      </StatusAlertButton>
    </CombinationGenerator>
    <Layout margin={{ bottom: 2 }}>
      <StatusAlertButton
        icon={SVGAsset.Check}
        statusAlertIcon={SVGAsset.Trash}
        statusAlertText="Remove"
      >
        Added
      </StatusAlertButton>
    </Layout>
    <Layout>
      <StatusAlertButton statusAlertText="Unfriend">Friend</StatusAlertButton>
    </Layout>
  </>
);
