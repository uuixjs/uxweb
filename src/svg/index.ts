export * from "./component";
export * from "./paths";
