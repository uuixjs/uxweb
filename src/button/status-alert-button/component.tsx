import { CoreButton, CoreButtonPublicProps, CoreButtonSize, CoreButtonType } from "../core-button";
import {
  FOCUS_VISIBLE_ATTR,
  css,
  focusVisible,
  hoverCss,
  hoverCssWithSelector,
  staticTokenRule,
  styleVariant,
  styled,
  themeTokenRule,
} from "@uuixjs/uuixweb-lib";
import { ForwardRefRenderFunction, ReactNode, forwardRef } from "react";
import { REVERSE_BUTTON_SIZES, ScButtonIconFigure } from "../button-icon/component";

import { CoreButtonLabel } from "../core-button/core-button-label";
import { SVGAsset } from "../../svg";

export interface StatusAlertButtonProps extends CoreButtonPublicProps {
  children?: string | ReactNode;
  icon?: SVGAsset;
  showAlert?: boolean;
  /** Icon to display when the status button is hovered. */
  statusAlertIcon?: SVGAsset;
  /**
   * Text to display when the status button is hovered. Setting this property
   * turns the button into a "status" button.
   */
  statusAlertText?: string;
}

const statusAlertActiveRules = css`
  background-color: ${themeTokenRule("color-background-button-status-active")};
  color: ${themeTokenRule("color-text-button-alert-active")};
`;

const statusAlertHoverFocusRules = css`
  background-color: ${themeTokenRule("color-background-button-status-hover")};
  color: ${themeTokenRule("color-text-button-alert")};
  &:active {
    ${statusAlertActiveRules}
  }
`;

interface ScStatusAlertButtonProps {
  $showAlert?: StatusAlertButtonProps["showAlert"];
  $iconOnly?: boolean;
}

const ScStatusAlertButton = styled(CoreButton)<Omit<StatusAlertButtonProps, "showAlert"> & ScStatusAlertButtonProps>`
  width: ${({ $iconOnly }) =>
    $iconOnly &&
    styleVariant("size", {
      [CoreButtonSize.Small]: staticTokenRule("button-size-small"),
      [CoreButtonSize.Default]: staticTokenRule("button-size-default"),
      [CoreButtonSize.Large]: staticTokenRule("button-size-large"),
    })};

  ${(props) =>
    !props.disabled &&
    css`
      transition-duration: ${staticTokenRule("timing-short")};
      transition-property: background-color, color;
      transition-timing-function: ease;
      background-color: ${themeTokenRule("color-background-button-status")};
      color: ${themeTokenRule("color-text-button-success")};
      ${props.$showAlert ? statusAlertHoverFocusRules : undefined}
      ${hoverCss`${statusAlertHoverFocusRules}`}
      ${focusVisible`${statusAlertHoverFocusRules}`}
      &:active: {${statusAlertActiveRules}}
    `}
`;

interface ScStatusAlertLabelProps {
  disabled?: boolean;
  $showAlert?: StatusAlertButtonProps["showAlert"];
  size: CoreButtonSize | "default" | "small" | "large"; // TODO: https://jira.xarth.tv/browse/COREUI-3554
  type: "primary" | "alert";
}

const buttonSizeValue = styleVariant<ScStatusAlertLabelProps, "size">("size", {
  [CoreButtonSize.Small]: staticTokenRule("button-size-small"),
  [CoreButtonSize.Default]: staticTokenRule("button-size-default"),
  [CoreButtonSize.Large]: staticTokenRule("button-size-large"),
});

const statusAlertLabelHoverFocusRules = (type: string) => {
  return `opacity: ${type === "alert" ? 1 : 0};`;
};

export const ScStatusAlertLabel = styled.div<ScStatusAlertLabelProps>`
  transition-duration: ${staticTokenRule("timing-short")};
  transition-property: opacity;
  transition-timing-function: ease;
  display: flex;
  align-items: center;
  justify-content: center;
  height: ${buttonSizeValue};

  ${(props) => {
    if (props.type === "alert") {
      return `
        opacity: 0;
        margin-top: calc(-1 * ${buttonSizeValue(props)});
      `;
    } else {
      return `
        opacity: 1;
        margin-top: 0;
      `;
    }
  }}

  ${(props) =>
    !props.disabled &&
    css`
      ${hoverCssWithSelector(`${ScStatusAlertButton}:hover &`)`
        ${statusAlertLabelHoverFocusRules(props.type)}
      `}

      ${props.$showAlert ? statusAlertLabelHoverFocusRules(props.type) : undefined}

      ${ScStatusAlertButton}${FOCUS_VISIBLE_ATTR} & {
        ${statusAlertLabelHoverFocusRules(props.type)}
      }
    `};
`;

const StatusAlertButton: ForwardRefRenderFunction<HTMLElement, StatusAlertButtonProps> = (
  {
    children,
    disabled,
    icon,
    showAlert,
    size = CoreButtonSize.Default,
    statusAlertIcon,
    statusAlertText,
    ...buttonProps
  },
  ref,
) => {
  return (
    <ScStatusAlertButton
      disabled={disabled}
      variant={disabled ? CoreButtonType.Primary : undefined}
      ref={ref}
      size={size}
      $showAlert={showAlert}
      $iconOnly={!buttonProps.fullWidth && !children && !!icon && !!statusAlertIcon}
      className="tw-status-alert-button"
      {...buttonProps}
    >
      <div>
        <ScStatusAlertLabel
          disabled={disabled}
          $showAlert={showAlert}
          size={size || CoreButtonSize.Default}
          type="primary"
        >
          {children ? (
            <CoreButtonLabel icon={icon} size={size} fullWidth={buttonProps.fullWidth || !!statusAlertIcon}>
              {children}
            </CoreButtonLabel>
          ) : (
            icon && <ScButtonIconFigure icon={icon} size={REVERSE_BUTTON_SIZES[size || CoreButtonSize.Default]} />
          )}
        </ScStatusAlertLabel>
        <ScStatusAlertLabel
          $showAlert={showAlert}
          disabled={disabled}
          size={size || CoreButtonSize.Default}
          type="alert"
        >
          {children || statusAlertText ? (
            <CoreButtonLabel icon={statusAlertIcon} size={size} fullWidth={buttonProps.fullWidth || !!statusAlertIcon}>
              {statusAlertText || children}
            </CoreButtonLabel>
          ) : (
            statusAlertIcon && (
              <ScButtonIconFigure icon={statusAlertIcon} size={REVERSE_BUTTON_SIZES[size || CoreButtonSize.Default]} />
            )
          )}
        </ScStatusAlertLabel>
      </div>
    </ScStatusAlertButton>
  );
};

StatusAlertButton.displayName = "StatusAlertButton";
const ComponentWithRef = forwardRef(StatusAlertButton);
export { ComponentWithRef as StatusAlertButton };
