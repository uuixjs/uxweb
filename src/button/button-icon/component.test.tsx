import { mount, shallow } from "enzyme";
import { createRef } from "react";
import { setupShallowTest } from "../../../tests/helpers";
import { CoreText } from "../../core-text";
import { Icon } from "../../icon";
import { SVGAsset } from "../../svg";
import { StatusAlertButton } from "../status-alert-button";
import {
  ButtonIcon,
  ButtonIconComponent,
  ButtonIconProps,
  ButtonIconSize,
  ButtonIconType,
} from "./component";

function getRequiredProps(): ButtonIconProps {
  return {
    "aria-label": "this is my aria label",
    icon: SVGAsset.Gear,
  };
}

function getOptionalProps(): ButtonIconProps {
  return {
    "aria-label": "this is my aria label",
    autoFocus: true,
    disabled: true,
    download: "Download all the things",
    dropdown: true,
    linkTo: "/",
    icon: SVGAsset.Gear,
    onClick: jest.fn(),
    overlay: true,
    size: ButtonIconSize.Large,
    tabIndex: 0,
    targetBlank: true,
    title: "test title",
    variant: ButtonIconType.Secondary,
  };
}

const setupShallow = setupShallowTest(ButtonIconComponent, getRequiredProps);
describe("ButtonIcon", () => {
  it("renders button icon component with required props", () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders button icon component with all props", () => {
    const { wrapper } = setupShallow(getOptionalProps());
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a button icon with a JSX.Element as icon", () => {
    const wrapper = mount(
      <ButtonIcon
        aria-label="element test"
        icon={<CoreText>JSX Element</CoreText>}
      />,
    );
    expect(wrapper.find(CoreText).length).toBe(1);
  });

  it("renders a button icon with as a StatusAlertButton", () => {
    const wrapper = mount(
      <ButtonIcon
        aria-label="element test"
        icon={SVGAsset.Heart}
        statusAlertIcon={SVGAsset.Unheart}
      />,
    );
    expect(wrapper.find(StatusAlertButton).length).toBe(1);
  });

  it("renders a button icon with an SVGAsset passed as an icon", () => {
    const wrapper = mount(
      <ButtonIcon aria-label="gear test" icon={SVGAsset.Gear} />,
    );
    expect(wrapper.find(Icon).length).toBe(1);
    expect(wrapper.find(Icon).prop("asset")).toEqual(SVGAsset.Gear);
  });

  it("renders a large button icon component", () => {
    const { wrapper } = setupShallow({ size: ButtonIconSize.Large });
    expect(wrapper).toMatchSnapshot();
  });

  it("renders button icon component as <a>", () => {
    const { wrapper } = setupShallow({ linkTo: "https://www.google.com" });
    expect(wrapper).toMatchSnapshot();
  });

  it("renders button icon component <button> with an onclick", () => {
    const { wrapper } = setupShallow({ onClick: jest.fn() });
    expect(wrapper).toMatchSnapshot();
  });

  it("renders button icon component as Link component", () => {
    const { wrapper } = setupShallow({ linkTo: "/dashbaord" });
    expect(wrapper).toMatchSnapshot();
  });

  it("passes the aria props to the html", () => {
    const wrapper = shallow(
      <ButtonIcon role="button" {...getRequiredProps()} />,
    );
    const html = wrapper.html();
    expect(html.match(/aria-label="this is my aria label"/)?.length).toBe(1);
    expect(html.match(/role="button"/)?.length).toBe(1);
  });
  it("passes the data props to the html", () => {
    const wrapper = mount(
      <ButtonIcon data-test-selector="test-selector" {...getRequiredProps()} />,
    );

    expect(wrapper.find("button")).toHaveProp({
      "data-test-selector": "test-selector",
    });
  });

  it("passes the ref to the CoreButton case", () => {
    const testRef = createRef<HTMLElement>();
    mount(<ButtonIcon ref={testRef} {...getRequiredProps()} />);
    expect(testRef.current).not.toBe(null);
  });

  it("passes the ref to the CoreInteractive case", () => {
    const testRef = createRef<HTMLElement>();
    mount(<ButtonIcon ref={testRef} {...getRequiredProps()} dropdown />);
    expect(testRef.current).not.toBe(null);
  });

  it("passes through the type attribute to the HTML button element", () => {
    const wrapper = mount(<ButtonIcon {...getRequiredProps()} type="button" />);

    expect(wrapper.find("button").prop("type")).toEqual("button");
  });
});
