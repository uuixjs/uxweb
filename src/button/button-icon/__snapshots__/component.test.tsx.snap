// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon renders a large button icon component 1`] = `
<Styled(CoreButton)
  aria-label="this is my aria label"
  size="large"
>
  <styled.div
    icon="Gear"
    size="large"
  />
</Styled(CoreButton)>
`;

exports[`ButtonIcon renders button icon component <button> with an onclick 1`] = `
<Styled(CoreButton)
  aria-label="this is my aria label"
  onClick={[MockFunction]}
  size="default"
>
  <styled.div
    icon="Gear"
  />
</Styled(CoreButton)>
`;

exports[`ButtonIcon renders button icon component as <a> 1`] = `
<Styled(CoreButton)
  aria-label="this is my aria label"
  linkTo="https://www.google.com"
  size="default"
>
  <styled.div
    icon="Gear"
  />
</Styled(CoreButton)>
`;

exports[`ButtonIcon renders button icon component as Link component 1`] = `
<Styled(CoreButton)
  aria-label="this is my aria label"
  linkTo="/dashbaord"
  size="default"
>
  <styled.div
    icon="Gear"
  />
</Styled(CoreButton)>
`;

exports[`ButtonIcon renders button icon component with all props 1`] = `
<Styled(CoreButton)
  $dropdown={true}
  aria-label="this is my aria label"
  autoFocus={true}
  disabled={true}
  download="Download all the things"
  linkTo="/"
  onClick={[MockFunction]}
  overlay={true}
  size="large"
  tabIndex={0}
  targetBlank={true}
  title="test title"
>
  <styled.div
    icon="Gear"
    size="large"
  />
  <styled.div
    icon="GlyphArrDown"
    size="large"
  />
</Styled(CoreButton)>
`;

exports[`ButtonIcon renders button icon component with required props 1`] = `
<Styled(CoreButton)
  aria-label="this is my aria label"
  size="default"
>
  <styled.div
    icon="Gear"
  />
</Styled(CoreButton)>
`;
